### PR TITLE
DEPR: Remove (Int|UInt|Float)64Index

### PR DESCRIPTION
**Pull request by topper-123**
Actually removes `Int64Index`, `UInt64Index` & `Float64Index`. Also does some cleanups that previous deprecation PRs didn't catch.  

This PR formally achieves the goals set out in #42717. In real terms there are a few outstanding items, which still need to be addressed:
* Move `NumericIndex` into `Index`
* Update docs and write the `whatsnew` entry for this change
* Make the tests in `pandas/tests/arithmetic` cover 8/16/32 bit dtypes
* a few minor issues

So I'd like to kep #42717 open until those are resolved.

### CODE REVIEW

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -498,7 +495,7 @@ def __new__(
 
         klass = cls._dtype_to_subclass(arr.dtype)
 
-        # _ensure_array _may_ be unnecessary once Int64Index etc are gone
+        # _ensure_array _may_ be unnecessary once NumericIndex etc are gone
```
Comment (jbrockmendel) [inline pandas/core/indexes/base.py:498]: i think changing this misses the point of this comment

File: pandas/compat/pickle_compat.py [hunk 2]
```diff
@@ -119,7 +119,7 @@ def load_reduce(self):
         "TimedeltaIndex",
     ),
     ("pandas.indexes.numeric", "Float64Index"): (
-        "pandas.core.indexes.numeric",
+        "pandas.core.indexes.base",
```
Comment (jbrockmendel) [inline pandas/compat/pickle_compat.py:122]: if these classes dont exist anymore, should we just remove these dict entries?

File: pandas/_testing/asserters.py [hunk 3]
```diff
@@ -75,8 +75,8 @@ def assert_almost_equal(
     right : object
     check_dtype : bool or {'equiv'}, default 'equiv'
         Check dtype if both a and b are the same type. If 'equiv' is passed in,
-        then `RangeIndex` and `Int64Index` are also considered equivalent
-        when doing type checking.
+        then `RangeIndex` and `NuericIndex` with int64 dtype are also considered
```
Comment (jbrockmendel) [inline pandas/_testing/asserters.py:78]: typo NuericIndex

File: pandas/core/indexes/base.py (hunk 1 above)
Comment (topper-123) [reply pandas/core/indexes/base.py:498]: I'm not sure I follow you here, could you expand?

File: pandas/compat/pickle_compat.py (hunk 2 above)
Comment (topper-123) [reply pandas/compat/pickle_compat.py:122]: These are a lot of other backward-compat here also, and I think there should be a common decision for all of these that are older than some determined release version. For example, remove everything older than v1.0.

I could ake a seperate issue, where that is discussed?

File: pandas/_testing/asserters.py (hunk 3 above)
Comment (topper-123) [reply pandas/_testing/asserters.py]: Fixed.

File: pandas/core/indexes/base.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/indexes/base.py:498]: the comment is suggesting that _ensure_array will become unnecessary

File: pandas/compat/pickle_compat.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/compat/pickle_compat.py:122]: > I could make a separate issue, where that is discussed?

@jreback any idea?  i think this is pickle-related, which i usually punt on

**Comment by topper-123** [comment]
The remaining failure look unrelated.

**Review by jbrockmendel — APPROVED** [review]
LGTM cc @mroeschke

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @topper-123

**Pull request merged**